feature-staking-impl/src/main/java/jp/co/soramitsu/staking/impl/presentation/staking/main/StakingViewModel.kt

```diff
@@ -110,12 +125,61 @@ class StakingViewModel @Inject constructor(
             it.alerts()
         }.distinctUntilChanged().shareIn(stakingStateScope, started = SharingStarted.Eagerly, replay = 1)
 
+    private val defaultNetworkInfoStates = mapOf(
+        StakingType.POOL to StakingAssetInfoViewState.StakingPool.default(resourceManager),
+        StakingType.RELAYCHAIN to StakingAssetInfoViewState.RelayChain.default(resourceManager),
+        StakingType.PARACHAIN to StakingAssetInfoViewState.Parachain.default(resourceManager)
+    )
+
+    private val stakingViewState1: SharedFlow<StakingViewState1?> = scenarioViewModelFlow
+        .flatMapLatest {
+            it.getStakingViewStateFlow1()
+        }.distinctUntilChanged().shareIn(stakingStateScope, started = SharingStarted.Eagerly, replay = 1)
+
+    inline fun <reified T : StakingAssetInfoViewState> Map<StakingType, StakingAssetInfoViewState>.get(type: StakingType): T = get(type) as T
+
+    private val networkInfoState: Flow<StakingAssetInfoViewState?> = networkInfo.map { networkInfoState ->
+        val selection = stakingSharedState.selectionItem.first()
+        if (selection.type != StakingType.POOL) return@map null
+        if (networkInfoState is LoadingState.Loaded) {
+            when (val state = networkInfoState.data) {
+                is StakingNetworkInfoModel.Parachain -> defaultNetworkInfoStates.get<StakingAssetInfoViewState.Parachain>(StakingType.PARACHAIN).update(state)
+                is StakingNetworkInfoModel.Pool -> defaultNetworkInfoStates.get<StakingAssetInfoViewState.StakingPool>(StakingType.POOL).update(state)
+                is StakingNetworkInfoModel.RelayChain -> defaultNetworkInfoStates.get<StakingAssetInfoViewState.RelayChain>(StakingType.RELAYCHAIN)
+                    .update(state)
+            }
+        } else {
+            defaultNetworkInfoStates[selection.type]!!
+        }
+    }.stateIn(scope = this, started = SharingStarted.Eagerly, initialValue = null)
+
+    val state = combine(
+        stakingSharedState.selectionItem,
+        assetSelectorMixin.selectedAssetModelFlow,
+        networkInfoState,
+        stakingViewState1
+    ) { selection, selectedAsset, networkInfo, stakingState ->
+
+        val selectorState = AssetSelectorState(
+            selectedAsset.tokenName,
+            selectedAsset.imageUrl,
+            selectedAsset.assetBalance,
+            (selectedAsset.selectionItem as? StakingAssetSelection.Pool)?.let { "pool" }
```

Comment:
why "pool", not something like just "(selectedAsset.selectionItem as? StakingAssetSelection.Pool)?.name" ?